common/djangoapps/student/models.py

```diff
@@ -1552,11 +1553,163 @@ def send_signal_full(cls, event, user=user, mode=mode, course_id=None, cost=None
                                   mode=mode, course_id=course_id,
                                   cost=cost, currency=currency)
 
-    def emit_event(self, event_name, enterprise_uuid=None):
+    @staticmethod
+    def _get_course_dates_for_email(blocks, user_timezone):
+        """
+        Getting nearest dates from today one would be before today and one
+        would be after today.
+        """
+
+        course_date_list = [
+            {
+                'id': 1,
+                'date': '',
+                'title': '',
+                'assignment_type': '',
+                'link': '',
+                'assignment_count': -1,
+                'due_date': '',
+                'due_time': ''
+            },
+            {
+                'id': 2,
+                'date': '',
+                'title': '',
+                'assignment_type': '',
+                'link': '',
+                'assignment_count': -1,
+                'due_date': '',
+                'due_time': ''
+            },
+            {
+                'id': 3,
+                'date': '',
+                'title': '',
+                'assignment_type': '',
+                'link': '',
+                'assignment_count': -1,
+                'due_date': '',
+                'due_time': ''
+            }
+        ]
```

Comment:
can we use a spread operator here instead of repetition for object attributes?
```
course_date = {
                'date': '',
                'title': '',
                'assignment_type': '',
                'link': '',
                'assignment_count': -1,
                'due_date': '',
                'due_time': ''
            }
course_date_list = [
            {
                'id': 1,
                **course_date,
            },
            {
                'id': 2,
                **course_date,
            },
            {
                'id': 3,
                **course_date,
            }
        ]
        
```